app/views/govuk_publishing_components/components/_print_link.html.erb

```diff
@@ -0,0 +1,11 @@
+<%
+  text ||= t('components.print_link.text')
+  href ||= "javascript:window.print()"
```

Comment:
This will currently raise a CSP warning (and soon it will be an error) as it generates in-line JavaScript. We don't use in-line JavaScript in the component library so I suggest we follow the current pattern and move this line into its own JavaScript file using the current module system (contextual-guidance, cookie-banner, and modal-dialogue are decent references for writing a JS module, but if things are not clear I'm more than happy to help).

Reply:
Ah righto - will do! Do you know what value I should set as the href? I read that without a href value some browsers don't treat anchor links as normal (but maybe that's not an issue). Also wasn't sure if href="#" was appropriate either.

Reply:
Might be worth not setting a default and raising an error/not rendering if no `href` is passed to the component?

Reply:
Could we just create this as a button rather than a link? The functionality is more like a button than a link anyway.

Reply:
Maybe worth considering for a future iteration.

Reply:
Ah, so I'm thinking of the case where there is no href and we want to rely entirely on JS.

Using just JS does make it seem more like a button - but then the implementation in government-frontend does use a href and makes sense to be a link.

Reply:
Ideally, we'd render either a `<button class="gem-c-print-link" data-module="print-link">Print entire page</button>` or a `<a class="gem-c-print-link" href="/print">Print entire page</a>` based on wether or not the component has an `href` passed in. If we're under pressure to deliver we can push it as is and raise and issue to clear debt later.

Reply:
Thanks! Yup - I think that makes sense, we aren't under enormous pressure so I'd to get it right and not add to the tech debt. I'm just seeing what design would be appropriate for the button - will update the PR and try to do the button or link implementation.

Reply:
We needed something similar ([a button styled as a link](https://github.com/alphagov/content-publisher/blob/master/app/assets/stylesheets/core/_link.scss#L1-L10)) in content-publisher, if that's of any help.